### ISSUE TITLE
Change case of modules/workflows to distinguish them from parameters

### ISSUE THREAD

**Issue by abhi18av** [fmalmeida/bacannot#38]
The current naming convention makes it a bit dubious to determine the exact nature of an identifier, for example in the snippet below 

```nextflow
      call_methylation(prokka.out[6])
      methylation_out_1 = call_methylation.out[2]
      methylation_out_2 = call_methylation.out[3]


```

Generally speaking, with DSL2, the community has adopted the paradigm of using `MODULE_NAME` / `WORKFLOW_NAME` pattern and regarding channels a preference is use a suffix like `ch`. Therefore the above snippet would be like 

```nextflow
      CALL_METHYLATION(PROKKA.out[6])
      methylation_out_1_ch = CALL_METHYLATION.out[2]
      methylation_out_2_ch = CALL_METHYLATION.out[3]
```

What are your thoughts?

**Comment by fmalmeida** [comment]
Yes, I agree that is better to distinguish modules/workflows to distinguish them from parameters. The pipeline would gain in readability, and standardization.

Please, also see my comments in #39 since it is related to this one.

**Comment by fmalmeida** [comment]
I have started to make these changes while trying to remodel the pipeline in branch https://github.com/fmalmeida/bacannot/tree/remodeling, related to issue #36 ... So we could try to address this issue in the same branch?

Or it is better to have a specific branch and specific PR for this issue and also for issue #39 ?

**Comment by abhi18av** [comment]
I think it's always better to have smaller PRs, this way if something goes wrong we know exactly where do we need to rollback.

**Comment by abhi18av** [comment]
I think that the changes introduced by https://github.com/fmalmeida/bacannot/pull/43 should not create big difference in the pipeline and once it has been tested on your end we can merge it on the `master`.

However, from now onwards, I am planning to follow this pattern 


1. Fork `abhinav/feature` from `develop`
2. Do my changes in `abhinav/feature` and create a PR against `develop`
3. Once they are merged (🤞) into `develop` we do further end testing
4. Then we create a release using a PR from `develop` to `master`.

Hopefully, this we can have visibility on what we are working on and have minimal merge conflicts.

Sounds good?

**Comment by fmalmeida** [comment]
Ok!

I will make the tests then in the `develop` branch to check if it could be merged.

And I totally agree with the plan you outlined. I will also try to make sure the `develop` branch is up-to-date with `master` to avoid conflicts when going to step 4.

**Comment by abhi18av** [comment]
Hey @fmalmeida , circling back for this one - let me know if there's anything I can help with :)

**Comment by fmalmeida** [comment]
Hi @abhi18av,

While trying to solve issue #36, I already started to change the casenames of channels and modules (which is related to the present issue). However, the issue #36 itself may be a little bit slow to come, but I am already changing it :)

Maybe it will be best to address this all there?

Or you believe it would be nice to solve this issue by itself, and then, when it's PR is merged to the `develop` I then try to bring its changes forward to the branch where issue #36 is being tackled?

Genuinely asking :) Don't know which one would be simpler or easier to manage and organise.

**Comment by abhi18av** [comment]
I'm of the thought that a PR should address a single issue and should be small in size. 

I noticed that there's a WIP branch https://github.com/fmalmeida/bacannot/tree/remodeling - could you please create a `draft` PR in the repo itself, it'd help in understanding the exact changes.


Beyond this, just to confirm you're in agreement with https://github.com/fmalmeida/bacannot/issues/38#issue-1055147985 right?

**Comment by fmalmeida** [comment]
Okay then. Let's make it that way:

1. We solve this issue here, by itself.
2. I will create a draft PR to keep track of changes related to issue #36 and its branch "remodeling"
3. I will stop making changes related to casename in issue #36 so it is easier to solve conflicts when bringing the changes of this issue here to the #36
4. Finally, when this issue is merged and its PR accepted I bring its changes to issue #36 so when it is finished, it does not raise too much conflicts with the changes found in the merged `develop`.

How does it sounds? Sounds ok?

:)

**Comment by abhi18av** [comment]
Sounds good to me @fmalmeida - thanks 😊

**Comment by fmalmeida** [comment]
This has been addressed by PR #45 

Thanks for the help @abhi18av

**Comment by abhi18av** [comment]
You're welcome Felipe! 😊
